diff -urN vmmon-only.org/include/compat_wait.h vmmon-only/include/compat_wait.h
--- vmmon-only.org/include/compat_wait.h	2004-05-12 16:49:05.725457240 +0200
+++ vmmon-only/include/compat_wait.h	2004-05-12 17:08:13.689940112 +0200
@@ -28,19 +28,7 @@
 
 #endif
 
-/*
- * The 'struct poll_wqueues' appeared in 2.5.48, when global
- * /dev/epoll interface was added.  It was backported to the
- * 2.4.20-wolk4.0s.
- */
-
-#if VMW_HAVE_EPOLL // {
 #define compat_poll_wqueues struct poll_wqueues
-#else // } {
-#define compat_poll_wqueues poll_table
-#endif // }
-
-#if VMW_HAVE_EPOLL // {
 
 /* If prototype does not match, build will abort here */
 extern void poll_initwait(compat_poll_wqueues *);
@@ -54,55 +42,4 @@
    poll_freewait((table)) \
 )
 
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2, 4, 0) // {
-
-/* If prototype does not match, build will abort here */
-extern void poll_initwait(compat_poll_wqueues *);
-
-#define compat_poll_initwait(wait, table) ( \
-   (wait) = (table), \
-   poll_initwait(wait) \
-)
-
-#define compat_poll_freewait(wait, table) ( \
-   poll_freewait((table)) \
-)
-
-#else // } {
-
-#define compat_poll_initwait(wait, table) ( \
-   (wait) = (table), /* confuse compiler */ \
-   (wait) = (poll_table *) __get_free_page(GFP_KERNEL), \
-   (wait)->nr = 0, \
-   (wait)->entry = (struct poll_table_entry *)((wait) + 1), \
-   (wait)->next = NULL \
-)
-
-static INLINE void
-poll_freewait(poll_table *wait)
-{
-   while (wait) {
-      struct poll_table_entry * entry;
-      poll_table *old;
-
-      entry = wait->entry + wait->nr;
-      while (wait->nr > 0) {
-	 wait->nr--;
-	 entry--;
-	 remove_wait_queue(entry->wait_address, &entry->wait);
-	 compat_fput(entry->filp);
-      }
-      old = wait;
-      wait = wait->next;
-      free_page((unsigned long) old);
-   }
-}
-
-#define compat_poll_freewait(wait, table) ( \
-   poll_freewait((wait)) \
-)
-
-#endif // }
-
-
 #endif /* __COMPAT_WAIT_H__ */
diff -urN vmnet-only.org/compat_sock.h vmnet-only/compat_sock.h
--- vmnet-only.org/compat_sock.h	2004-05-12 16:49:09.092945304 +0200
+++ vmnet-only/compat_sock.h	2004-05-12 17:09:13.732812208 +0200
@@ -5,20 +5,4 @@
 #include <net/sock.h>
 
 
-/*
- * Between 2.5.70 and 2.5.71 all sock members were renamed from XXX to sk_XXX.
- *
- * VMW_HAVE_SK_WMEM_ALLOC is defined in module Makefile if kernel's struct sock
- * has sk_wmem_alloc member. See vmnet's Makefile.kernel for details.
- * It also means that all modules including this file should do
- *
- * EXTRA_CFLAGS += $(call vm_check_build, $(SRCROOT)/socket.c,  -DVMW_HAVE_SK_WMEM_ALLOC, )
- *
- * in their Makefiles.
- */
-#ifndef VMW_HAVE_SK_WMEM_ALLOC
-#   define sk_wmem_alloc wmem_alloc
-#endif
-
-
 #endif /* __COMPAT_SOCK_H__ */
diff -urN vmnet-only.org/compat_wait.h vmnet-only/compat_wait.h
--- vmnet-only.org/compat_wait.h	2004-05-12 16:49:09.092945304 +0200
+++ vmnet-only/compat_wait.h	2004-05-12 17:08:13.691939808 +0200
@@ -28,19 +28,7 @@
 
 #endif
 
-/*
- * The 'struct poll_wqueues' appeared in 2.5.48, when global
- * /dev/epoll interface was added.  It was backported to the
- * 2.4.20-wolk4.0s.
- */
-
-#if VMW_HAVE_EPOLL // {
 #define compat_poll_wqueues struct poll_wqueues
-#else // } {
-#define compat_poll_wqueues poll_table
-#endif // }
-
-#if VMW_HAVE_EPOLL // {
 
 /* If prototype does not match, build will abort here */
 extern void poll_initwait(compat_poll_wqueues *);
@@ -54,55 +42,4 @@
    poll_freewait((table)) \
 )
 
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(2, 4, 0) // {
-
-/* If prototype does not match, build will abort here */
-extern void poll_initwait(compat_poll_wqueues *);
-
-#define compat_poll_initwait(wait, table) ( \
-   (wait) = (table), \
-   poll_initwait(wait) \
-)
-
-#define compat_poll_freewait(wait, table) ( \
-   poll_freewait((table)) \
-)
-
-#else // } {
-
-#define compat_poll_initwait(wait, table) ( \
-   (wait) = (table), /* confuse compiler */ \
-   (wait) = (poll_table *) __get_free_page(GFP_KERNEL), \
-   (wait)->nr = 0, \
-   (wait)->entry = (struct poll_table_entry *)((wait) + 1), \
-   (wait)->next = NULL \
-)
-
-static INLINE void
-poll_freewait(poll_table *wait)
-{
-   while (wait) {
-      struct poll_table_entry * entry;
-      poll_table *old;
-
-      entry = wait->entry + wait->nr;
-      while (wait->nr > 0) {
-	 wait->nr--;
-	 entry--;
-	 remove_wait_queue(entry->wait_address, &entry->wait);
-	 compat_fput(entry->filp);
-      }
-      old = wait;
-      wait = wait->next;
-      free_page((unsigned long) old);
-   }
-}
-
-#define compat_poll_freewait(wait, table) ( \
-   poll_freewait((wait)) \
-)
-
-#endif // }
-
-
 #endif /* __COMPAT_WAIT_H__ */
